docs/generator.py

```diff
@@ -0,0 +1,43 @@
+from pytablewriter import MarkdownTableWriter
+from pytablewriter.style import Style
+
+import jupyterhub.metrics
+
+
+class Generator:
+    @classmethod
+    def create_writer(cls, table_name, headers, values):
+        writer = MarkdownTableWriter()
+        writer.table_name = table_name
+        writer.headers = headers
+        writer.value_matrix = values
+        writer.margin = 1
+        [writer.set_style(header, Style(align="center")) for header in headers]
+        return writer
+
+    def _parse_metrics(self):
+        table_rows = []
+        for name in dir(jupyterhub.metrics):
+            obj = getattr(jupyterhub.metrics, name)
+            if obj.__class__.__module__.startswith('prometheus_client.'):
+                for metric in obj.describe():
+                    table_rows.append([metric.type, metric.name, metric.documentation])
+        return table_rows
+
+    def prometheus_metrics(self):
+        filename = "./source/monitoring/metrics.md"
+        table_name = "List of Prometheus Metrics\n"
```

Comment:
Should the table name include a newline?

Reply:
Without it there was no empty line b/w the Title and the content.
I've removed it for now. Will search for better alternatives.